src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -1234,7 +1236,16 @@ namespace xclhwemhal2 {
 
            const char *curr = (const char *)hostBuf;
            //Note: Adding PF and BAR ID valuesas 0, Once original values are avaiaable they get replaced
-           xclWriteAddrSpaceDeviceRam_RPC_CALL(xclWriteAddrSpaceDeviceRam ,space,offset,curr,size,0,0);
+           uint64_t pf_id = 0;
+           uint64_t bar_id = 0;
+           for(uint32_t i = 0 ;i < pf_bar_info.size(); i++) {
+               if(offset >= pf_bar_info[i]->remap_addr &&  ( offset <= pf_bar_info[i]->remap_addr + pf_bar_info[i]->size ) ) {
+                   pf_id = pf_bar_info[i]->pf_id;
+                   bar_id = pf_bar_info[i]->bar_id;
+                   break;
+               }
+           }
```

Comment:
This is duplicated in several places. Can you create a function and use it.

Reply:
I have added member function for repeated logic